### PR TITLE
Qdrant integration tutorial

### PR DESCRIPTION
**Pull request by merobi-hub**
This adds a Learn tutorial based on [Semantic querying with Airflow and Astronomer](https://qdrant.tech/documentation/examples/qdrant-airflow-astronomer/).

### CODE REVIEW

File: learn/airflow-qdrant.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+`embed_description`: This is a dynamic task that creates one mapped task instance for each book in the list. The task uses the `embed` function to generate vector embeddings for each description. To use a different embedding model, you can adjust the `EMBEDDING_MODEL_ID` and `EMBEDDING_DIMENSION` values.
+
+`embed_user_preference`: Here, we take a user’s input and convert it into a vector using the same pre-trained model used for the book descriptions.
+
+`qdrant_vector_ingest`: This task ingests the book data into the Qdrant collection using the `QdrantIngestOperator`, associating each book description with its corresponding vector embeddings.
+
+`search_qdrant`: Finally, this task performs a search in the Qdrant database using the vectorized user preference. It finds the most relevant book in the collection based on vector similarity.
+
+### Run the DAG
```
Comment (kentdanas) [inline learn/airflow-qdrant.md:214]:
```suggestion
## Step 5: Run your DAG
```

File: learn/airflow-qdrant.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+Head over to your terminal and run `astro dev start`.
```
Comment (kentdanas) [inline learn/airflow-qdrant.md:216]:
```suggestion
1. Run `astro dev start` in your Astro project to start Airflow and open the Airflow UI at `localhost:8080`.
```

File: learn/airflow-qdrant.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+A local Airflow container should spawn. You can now access the Airflow UI at http://localhost:8080. Visit our DAG by clicking on **books_recommend**.
```
Comment (kentdanas) [inline learn/airflow-qdrant.md:218]:
```suggestion
2. In the Airflow UI, run the `books_recommend` DAG by clicking the play button. You'll be asked for input about your book preference.
```

File: learn/airflow-qdrant.md [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+![Qdrant demo DAG](/img/integrations/qdrant-demo-dag.png)
+
+Hit the **PLAY** button on the right to run the DAG. You’ll be asked for input about your preference, with the default value already filled in.
```
Comment (kentdanas) [inline learn/airflow-qdrant.md:222]:
```suggestion
```

File: learn/airflow-qdrant.md [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+`search_qdrant`: Finally, this task performs a search in the Qdrant database using the vectorized user preference. It finds the most relevant book in the collection based on vector similarity.
+
+### Run the DAG
+
+Head over to your terminal and run `astro dev start`.
+
+A local Airflow container should spawn. You can now access the Airflow UI at http://localhost:8080. Visit our DAG by clicking on **books_recommend**.
+
+![Qdrant demo DAG](/img/integrations/qdrant-demo-dag.png)
```
Comment (kentdanas) [inline learn/airflow-qdrant.md:220]: Let's move this screenshot up into the Write the DAG section where you describe the tasks.

File: learn/airflow-qdrant.md [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+
+Hit the **PLAY** button on the right to run the DAG. You’ll be asked for input about your preference, with the default value already filled in.
+
+![Qdrant reference input](/img/integrations/qdrant-reference-input.png)
+
+After your DAG run completes, you should be able to see the output of your search in the logs of the `search_qdrant` task.
```
Comment (kentdanas) [inline learn/airflow-qdrant.md:226]:
```suggestion
3. View the output of your search in the logs of the `search_qdrant` task.
```

File: learn/airflow-qdrant.md [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+
+![Qdrant output](/img/integrations/qdrant-output.png)
+
+There you have it, an Airflow pipeline that interfaces with Qdrant! Feel free to fiddle around and explore Airflow. The following references might come in handy.
```
Comment (kentdanas) [inline learn/airflow-qdrant.md:230]:
```suggestion
```

File: learn/airflow-qdrant.md [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+
+## Further reading
+
+- [Introduction to Airflow](https://docs.astronomer.io/learn/intro-to-airflow)
+- [Airflow Concepts](https://docs.astronomer.io/learn/category/airflow-concepts)
+- [Airflow Reference](https://airflow.apache.org/docs/)
+- [Astronomer Documentation](https://docs.astronomer.io/)
```
Comment (kentdanas) [inline learn/airflow-qdrant.md:237]:
```suggestion
```

**Review by kentdanas — CHANGES_REQUESTED** [review]
@merobi-hub thanks for this! Suggestions are all to bring it in line with the formatting and language we use for our other integration tutorials. You may need to double check indentation and numbering, but otherwise should be straightforward.

File: learn/airflow-qdrant.md [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+
+- A running Qdrant instance. A [free instance](https://cloud.qdrant.io) is available.
+- The Astro CLI. See [Install the Astro CLI](https://docs.astronomer.io/astro/cli/install-cli).
+- A [HuggingFace token](https://huggingface.co/docs/hub/en/security-tokens) to generate embeddings.
+
+## Step 1: Set up the project
+
+1. Create a new Astro project:
+
+```bash
```
Comment (kentdanas) [inline learn/airflow-qdrant.md:31]: If you tab these in they'll render in line with the numbered steps

File: learn/airflow-qdrant.md [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+```text
+HUGGINGFACE_TOKEN="<YOUR_HUGGINGFACE_ACCESS_TOKEN>"
+AIRFLOW_CONN_QDRANT_DEFAULT='{
+    "conn_type": "qdrant",
+    "host": "xyz-example.eu-central.aws.cloud.qdrant.io:6333",
+    "password": "<YOUR_QDRANT_API_KEY>"
+}'
+```
+
+## Step 3: Add the data corpus
```
Comment (kentdanas) [inline learn/airflow-qdrant.md:55]:
```suggestion
## Step 3: Add your data
```

File: learn/airflow-qdrant.md [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+
+2. To use Qdrant within Airflow, install the Qdrant Airflow provider by adding the following to your `requirements.txt` file:
+
+```text
+apache-airflow-providers-qdrant==1.1.0
+```
+
+## Step 2: Configure credentials
+
+Add the following configuration to your `.env` file to create Airflow connections between Airflow and HuggingFace and Qdrant.
```
Comment (kentdanas) [inline learn/airflow-qdrant.md:44]:
```suggestion
Add the following to your `.env` file to create Airflow connections between Airflow and HuggingFace and Qdrant. Make sure to update with your HuggingFace access token and Qdrant instance details.
```

**Review by kentdanas — APPROVED** [review]
Looks great @merobi-hub! A few very minor comments, otherwise good to have this copy edited. @jwitz @lzdanski if one of you could take a quick look that would be awesome :)

File: learn/airflow-qdrant.md [hunk 12]
```diff
@@ -0,0 +1,219 @@
...
+sidebar_custom_props: { icon: 'img/integrations/qdrant.png' }
+---
+
+import CodeBlock from '@theme/CodeBlock';
+
+[Qdrant](https://qdrant.tech/) is an open-source vector database and similarity search engine designed for AI applications.
+
+In this tutorial, you'll use the [Qdrant Airflow provider](https://airflow.apache.org/docs/apache-airflow-providers-qdrant/stable/index.html) to write a DAG that generates embeddings in parallel and performs semantic retrieval based on user input.
+
+Airflow is useful when running operations in Qdrant based on data events or building parallel tasks for generating vector embeddings. By using Airflow, you can set up monitoring and alerts for your pipelines for full observability.
```
Comment (lzdanski) [inline learn/airflow-qdrant.md:15]:
```suggestion
Airflow provides useful operational and orchestration when running operations in Qdrant based on data events or building parallel tasks for generating vector embeddings. By using Airflow, you can set up monitoring and alerts for your pipelines for full observability.
```

something like that? shift from active voice and a little more specific in relating the qdrant operations to Airflow options?

File: learn/airflow-qdrant.md [hunk 13]
```diff
@@ -0,0 +1,219 @@
...
+
+## Time to complete
+
+This tutorial takes approximately 30 minutes to complete.
+
+## Prerequisites
+
+- A running Qdrant instance. A [free instance](https://cloud.qdrant.io) is available.
+- The Astro CLI. See [Install the Astro CLI](https://docs.astronomer.io/astro/cli/install-cli).
+- A [HuggingFace token](https://huggingface.co/docs/hub/en/security-tokens) to generate embeddings.
```
Comment (lzdanski) [inline learn/airflow-qdrant.md:25]: Everytime I see `huggingface` it makes me think of Alien.

File: learn/airflow-qdrant.md [hunk 14]
```diff
@@ -0,0 +1,219 @@
...
+## Step 1: Set up the project
+
+1. Create a new Astro project:
+
+    ```bash
+    mkdir qdrant-airflow-tutorial && cd qdrant-airflow-tutorial
+    astro dev init
+    ```
+
+2. To use Qdrant within Airflow, install the Qdrant Airflow provider by adding the following to your `requirements.txt` file:
```
Comment (lzdanski) [inline learn/airflow-qdrant.md:36]:
```suggestion
2. To use Qdrant in Airflow, install the Qdrant Airflow provider by adding the following to your `requirements.txt` file:
```

File: learn/airflow-qdrant.md [hunk 15]
```diff
@@ -0,0 +1,219 @@
...
+
+        ```text
+    apache-airflow-providers-qdrant==1.1.0
+    ```
```
Comment (lzdanski) [inline learn/airflow-qdrant.md:40]:
```suggestion

    ```text
    apache-airflow-providers-qdrant==1.1.0
    ```
```

File: learn/airflow-qdrant.md [hunk 16]
```diff
@@ -0,0 +1,219 @@
...
+
+## Step 2: Configure credentials
+
+Add the following to your `.env` file to create Airflow connections between Airflow and HuggingFace and Qdrant. Make sure to update with your HuggingFace access token and Qdrant instance details.
```
Comment (lzdanski) [inline learn/airflow-qdrant.md:44]:
```suggestion
Add the following code to your `.env` file to create Airflow connections between Airflow and HuggingFace and Qdrant. Make sure to update the sample code with your HuggingFace access token and Qdrant instance details.
```

File: learn/airflow-qdrant.md [hunk 17]
```diff
@@ -0,0 +1,219 @@
...
+    AIRFLOW_CONN_QDRANT_DEFAULT='{
+        "conn_type": "qdrant",
+        "host": "xyz-example.eu-central.aws.cloud.qdrant.io:6333",
+        "password": "<YOUR_QDRANT_API_KEY>"
+    }'
+    ```
+
+## Step 3: Add your data
+
+Paste the following sample data into a file called `books.txt` within your `include` directory.
```
Comment (lzdanski) [inline learn/airflow-qdrant.md:57]:
```suggestion
Paste the following sample data into a file called `books.txt` in your `include` directory.
```

File: learn/airflow-qdrant.md [hunk 18]
```diff
@@ -0,0 +1,219 @@
...
+    8 | The Lord of the Rings (1954-1955) | fantasy | J.R.R. Tolkien's epic fantasy trilogy follows the journey of Frodo Baggins to destroy the One Ring and defeat the Dark Lord Sauron in the land of Middle-earth.
+    9 | The Alchemist (1988) | fiction | Paulo Coelho's philosophical novel follows Santiago, an Andalusian shepherd boy, on a journey of self-discovery and spiritual awakening as he searches for a hidden treasure.
+    10 | The Da Vinci Code (2003) | mystery/thriller | Dan Brown's gripping thriller follows symbologist Robert Langdon as he unravels clues hidden in art and history while trying to solve a murder mystery with far-reaching implications.
+    ```
+
+## Step 4: Create your DAG
+
+1. In your `dags` folder, create a file called `books_recommend.py`.
+
+2. Copy the following code into the file.
```
Comment (lzdanski) [inline learn/airflow-qdrant.md:76]:
```suggestion
2. Copy the following code into the `books_recommend.py` file.
```

File: learn/airflow-qdrant.md [hunk 19]
```diff
@@ -0,0 +1,219 @@
...
+
+    ```python
```
Comment (lzdanski) [inline learn/airflow-qdrant.md:78]: You might want to put this code into the `details` formatting feature. It lets users expand/hide the code examples in a formatted box. The end of the `Hosted worker types` docs section has one - https://www.astronomer.io/docs/astro/configure-worker-queues#hosted-worker-types

```
<details>
<strong> Recommend books DAG</strong>

code
</details>
```

File: learn/airflow-qdrant.md [hunk 20]
```diff
@@ -0,0 +1,219 @@
...
+            qdrant_vector_ingest,
+            search_qdrant(embed_preference()),
+        )
+
+
+    recommend_book()
+
+    ```
+
+This DAG consists of 6 tasks that generate embeddings in parallel for the data corpus and perform semantic retrieval based on user input.
```
Comment (lzdanski) [inline learn/airflow-qdrant.md:194]:
```suggestion
This QDrant Demo DAG consists of six tasks that generate embeddings in parallel for the data corpus and perform semantic retrieval based on user input.
```
numbers fewer than 10 are colloquially spelled out.

File: learn/airflow-qdrant.md [hunk 21]
```diff
@@ -0,0 +1,219 @@
...
+- `import_books`: This task reads a text file containing information about the books (such as title, genre, and description) and then returns the data as a list of dictionaries.
+
+- `init_collection`: This task initializes a collection in the Qdrant database, where we will store the vector representations of the book descriptions. The `recreate_collection()` function deletes a collection first if it already exists. Trying to create a collection that already exists throws an error.
```
Comment (lzdanski) [inline learn/airflow-qdrant.md:197]:
```suggestion
- `init_collection`: This task initializes a collection in the Qdrant database, where you store the vector representations of the book descriptions. The `recreate_collection()` function deletes a collection first if it already exists. Trying to create a collection that already exists throws an error.
```

File: learn/airflow-qdrant.md [hunk 22]
```diff
@@ -0,0 +1,219 @@
...
+
+- `embed_description`: This is a dynamic task that creates one mapped task instance for each book in the list. The task uses the `embed` function to generate vector embeddings for each description. To use a different embedding model, you can adjust the `EMBEDDING_MODEL_ID` and `EMBEDDING_DIMENSION` values.
+
+- `embed_user_preference`: Here, we take a user’s input and convert it into a vector using the same pre-trained model used for the book descriptions.
```
Comment (lzdanski) [inline learn/airflow-qdrant.md:201]:
```suggestion
- `embed_user_preference`: This task takes a user’s input and converts it into a vector using the same pre-trained model used for the book descriptions.
```

File: learn/airflow-qdrant.md [hunk 23]
```diff
@@ -0,0 +1,219 @@
...
+
+- `qdrant_vector_ingest`: This task ingests the book data into the Qdrant collection using the `QdrantIngestOperator`, associating each book description with its corresponding vector embeddings.
+
+- `search_qdrant`: Finally, this task performs a search in the Qdrant database using the vectorized user preference. It finds the most relevant book in the collection based on vector similarity.
+
+    ![Qdrant demo DAG](/img/integrations/qdrant-demo-dag.png)
```
Comment (lzdanski) [inline learn/airflow-qdrant.md:207]:
```suggestion
    ![Screenshot of the graph view for the Qdrant demo DAG, with the each of the six steps shown.](/img/integrations/qdrant-demo-dag.png)
```

File: learn/airflow-qdrant.md [hunk 24]
```diff
@@ -0,0 +1,219 @@
...
+
+## Step 5: Run your DAG
+
+1. Run `astro dev start` in your Astro project to start Airflow and open the Airflow UI at `localhost:8080`.
+
+2. In the Airflow UI, run the `books_recommend` DAG by clicking the play button. You'll be asked for input about your book preference.
+
+    ![Qdrant reference input](/img/integrations/qdrant-reference-input.png)
```
Comment (lzdanski) [inline learn/airflow-qdrant.md:215]:
```suggestion
    ![Qdrant reference input shows an example of the UI where Airflow prompts you to enter your book preference.](/img/integrations/qdrant-reference-input.png)
```

File: learn/airflow-qdrant.md [hunk 25]
```diff
@@ -0,0 +1,219 @@
...
+
+3. View the output of your search in the logs of the `search_qdrant` task.
+
+    ![Qdrant output](/img/integrations/qdrant-output.png)
```
Comment (lzdanski) [inline learn/airflow-qdrant.md:219]:
```suggestion
    ![Screenshot of the Qdrant output example, that shows the title and description of the book recommendation.](/img/integrations/qdrant-output.png)
```

**Review by lzdanski — CHANGES_REQUESTED** [review]

File: learn/airflow-qdrant.md [hunk 26]
```diff
@@ -0,0 +1,222 @@
...
+    8 | The Lord of the Rings (1954-1955) | fantasy | J.R.R. Tolkien's epic fantasy trilogy follows the journey of Frodo Baggins to destroy the One Ring and defeat the Dark Lord Sauron in the land of Middle-earth.
+    9 | The Alchemist (1988) | fiction | Paulo Coelho's philosophical novel follows Santiago, an Andalusian shepherd boy, on a journey of self-discovery and spiritual awakening as he searches for a hidden treasure.
+    10 | The Da Vinci Code (2003) | mystery/thriller | Dan Brown's gripping thriller follows symbologist Robert Langdon as he unravels clues hidden in art and history while trying to solve a murder mystery with far-reaching implications.
+    ```
+
+## Step 4: Create your DAG
+
+1. In your `dags` folder, create a file called `books_recommend.py`.
+
+2. Copy the following code into the `books_recommend.py` file.
```
Comment (lzdanski) [inline learn/airflow-qdrant.md:76]:
```suggestion
2. Copy the following **Recommend books DAG** code into the `books_recommend.py` file.
```

**Review by lzdanski — APPROVED** [review]
One small thing! I think this will help emphasize that the code is in the expandable section. Looks fantastic!

**Pull request merged**